sale_order_type/models/sale.py

```diff
@@ -99,6 +107,17 @@ def onchange_type_id(self):
             if line_vals:
                 order.order_line.update(line_vals)
 
+    @api.onchange("user_id")
+    def onchange_user_id(self):
+        res = super().onchange_user_id()
+        if (
+            self.state in ["draft", "sent"]
+            and not self.user_id.property_warehouse_id
+            and self.type_id.warehouse_id
```

Comment:
Not very sure about this. Should user preference have precedence over sale type options?

Reply:
Note that user_id is equivalent to salesman.

The option for a salesman to have a warehouse assigned to him is very very specific.
Normally salesmen don't have any assigned to them, but if they do this option should override the warehouse assigned to the order type.

Maybe you can find a case where you want a different behaviour, but at the moment the problem reported to us is this.

Anyway these are still default values that can be changed by the user.

Reply:
But this is not doing that, but overwriting whatever is already assigned, and put the picking type warehouse if not warehouse defined in the salesman (which will be mostly always). Moreover, this is an onchange, and not a computed method, so I think there's a mix of both around.

Reply:
I can think in both scenarios :sweat_smile: or even in mixed ones :smile: And it seems that the logic is inverted from the one defined avobe where the sale type warehouse has the precedence...

Reply:
Well, the problem is the above, can you make a better code suggestion to solve it?

Reply:
I've tried the sane option, but it doesn't solve the problem... :wink: 
Any other suggestions?

I repeat the objective:
The warehouse of the salesperson (if established) is much more specific than the order type warehouse.